src/components/Facets.tsx

```diff
@@ -22,67 +24,109 @@ enum FacetType {
   if (children) {
     (Array.isArray(children) ? children : [children])
-      .forEach(child => fieldIdToCustomFacetProps.set(child.props.fieldId, child));
+      .forEach(child => {
+        fieldIdToCustomFacetProps.set(child.props.fieldId, child);
+        fieldIds.push(child.props.fieldId);
+      });
   }
 
   return (
     <div>
       <FacetsProvider searchOnChange={searchOnChange} className={customCssClasses.facetsContainer}>
-        {facets => facets
-          .map((facet, i) => {
-            let facetType: FacetType = FacetType.STANDARD;
-            let facetProps: FacetProps = {
-              fieldId: facet.fieldId,
-              label: facet.displayName,
-            };
-            if (fieldIdToCustomFacetProps.has(facet.fieldId)) {
-              const customFacetElement: ReactElement = fieldIdToCustomFacetProps.get(facet.fieldId);
-              facetProps = customFacetElement.props;
-              facetType = getFacetTypeFromReactElementType(
-                (typeof customFacetElement.type === 'function') ? customFacetElement.type.name : '');
-            } else {
-              if (isStringFacet(facet)) {
-                facetType = FacetType.STANDARD;
-              } else if (isNumericalFacet(facet)) {
-                facetType = FacetType.NUMERICAL;
+        {facets => {
+          if (!facets || !facets.length) {
+            return;
+          }
+
+          if (!onlyRenderChildren) {
+            facets.forEach(facet => {
+              if (!fieldIds.includes(facet.fieldId)) {
+                fieldIds.push(facet.fieldId);
+              }
+            });
+          }
+
+          const fieldIdToFacet = new Map();
+          facets.forEach(facet => fieldIdToFacet.set(facet.fieldId, facet));
+
+          return fieldIds
+            .filter(fieldId =>
+              !excludedFieldIds.includes(fieldId)
+              && fieldIdToFacet.get(fieldId).options.length > 0
+              && (!onlyRenderChildren || fieldIdToCustomFacetProps.has(fieldId)))
+            .map((fieldId, i) => {
+              console.log("fieldId", fieldId);
+              const facet: DisplayableFacet = fieldIdToFacet.get(fieldId);
+              let facetType: FacetType = FacetType.STANDARD;
+              let facetProps: FacetProps = {
+                fieldId: facet.fieldId,
+                label: facet.displayName,
+              };
+              if (fieldIdToCustomFacetProps.has(facet.fieldId)) {
+                const customFacetElement: ReactElement =
+                  fieldIdToCustomFacetProps.get(facet.fieldId);
+                facetProps = { ...facetProps, ...customFacetElement.props };
+                facetType = getFacetTypeFromReactElementType(
+                  (typeof customFacetElement.type === 'function')
+                    ? customFacetElement.type.name : '');
+              } else {
+                if (isHierarchicalFacet(facet, delimiter)) {
+                  facetType = FacetType.HIERARCHICAL;
```

Comment:
is it possible to collapse this into a single if/switch statement?

Reply:
I moved this section to its own function so the main logic is clearer